src/main/java/org/isf/admission/rest/AdmissionController.java

```diff
@@ -332,15 +332,12 @@ public ResponseEntity<Integer> getUsedWardBed(@RequestParam("wardid") String war
 	@DeleteMapping(value = "/admissions/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<Boolean> deleteAdmissionType(@PathVariable("id") int id) throws OHServiceException {
 		LOGGER.info("setting admission to deleted: {}", id);
-		boolean isDeleted;
 		Admission admission = admissionManager.getAdmission(id);
-		if (admission != null) {
-			isDeleted = admissionManager.setDeleted(id);
-		} else {
+		if (admission == null) {
 			return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
 		}
-
-		return ResponseEntity.ok(isDeleted);
+		admissionManager.setDeleted(id);
+		return ResponseEntity.ok(true);
```

Comment:
what if the `setDeleted(id)` returns null?

Reply:
Up

Reply:
In this method it will never return `null` as lines 335-337 are:
<pre>
                Admission admission = admissionManager.getAdmission(id);
		if (admission == null) {
			return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
		}
</pre>
Which means if the admission with that `id` doesn't exist it never executes `admissionManager.setDeleted(id)` and returns`NOT_FOUND`.  The only way `setDeleted(id)` returns `null` is if the record doesn't exist but it is short circuited by the  `null` check in this file.

Reply:
sorry, I made an oversight